docs/GettingStarted/build.md

```diff
@@ -1,42 +1,97 @@
 
-## Dependencies
+# Building
+We mostly rely on a build container, but some dependencies are still needed outside that container: our fork of [LLVM20](https://github.com/trail-of-forks/clangir), a local copy of `lld`, and LLVM [LIT](https://llvm.org/docs/CommandGuide/lit.html).
 
-| Name | Version |
-| ---- | ------- |
-| [CMake](https://cmake.org/) | >= 3.25.0 |
-| [LLVM](http://llvm.org/) | == 18 |
+In order to set up those and build Patchestry, please follow the first-time instructions for your development environment of choice:
+- [macOS](#first-time-setup-macos)
+- [Linux](#first-time-setup-linux)
 
-Currently, it is necessary to use `clang` (due to `gcc` bug) to build Patchestry, because of `patchestry`. On Linux it is also necessary to use `lld` at the moment.
+See also: [Development](#development)
 
-Patchestry uses `llvm-18` which can be obtained from the [repository](https://apt.llvm.org/) provided by LLVM.
+# First Time Development Setup: MacOS
 
-Before building (for Ubuntu) get all the necessary dependencies by running
+1. Install Xcode from the App Store and set up the command line tools:
+   ```
+   xcode-select --install
+   ```
+
+2. Install the required dependencies using Homebrew:
+   ```
+   # basics
+   brew install colima docker docker-buildx docker-credential-helper cmake lit
+   # to run tests, we need to cross-compile for x86-64/Linux and aarch64/Linux
+   brew install FiloSottile/musl-cross/musl-cross
+   ```
+
+3. Configure Docker BuildX to work with Colima:
+   ```
+   mkdir -p ~/.docker/cli-plugins
+   ln -s $(which docker-buildx) ~/.docker/cli-plugins/docker-buildx
+   colima restart
+   docker buildx version
+   ```
+
+4. Log into Docker Hub:
```

Comment:
Do you need to log into docker hub? The images can be build locally.

Reply:
I didn't need to do anything with Docker Hub, but for Mac I think you do, so I left it. The Mac instructions are as they were, just moved!